test/uk/gov/hmrc/slacknotifications/services/NotificationServiceSpec.scala

```diff
@@ -78,6 +78,62 @@ class NotificationServiceSpec extends WordSpec with Matchers with ScalaFutures w
         result shouldBe NotificationResult().addError(SlackError(statusCode, errorMsg))
       }
     }
+
+    "return an error if an exception was thrown by the Slack connector" in new Fixtures {
+      val errorMsg = "some exception message"
+
+      val genExceptions =
+        Gen.oneOf(
+          new BadRequestException(errorMsg),
+          Upstream4xxResponse(errorMsg, 403, 403),
+          new NotFoundException(errorMsg),
+          Upstream5xxResponse(errorMsg, 500, 500))
+
+      forAll(genExceptions) { ex =>
+        when(slackConnector.sendMessage(any())(any()))
+          .thenReturn(Future.failed(ex))
+
+        val result = service
+          .sendSlackMessage(
+            SlackMessage(
+              channel     = "channel",
+              text        = "text",
+              username    = "someUser",
+              icon_emoji  = Some(":snowman:"),
+              attachments = Nil))
+          .futureValue
+
+        result shouldBe
```

Comment:
Logic of catching exceptions and converting to an error is the same in business code and the test.

Maybe this could be done as a table driven test, like below (pseudo code) but the end result is very similar

```
val exceptionsAndExpectedErrors = 
Table(
"exception" -> "expected problem",
HttpException -> slackError1,
...

)

forAll(exceptionsAndExpectedErrors) { case (ex, expectedError) =>
   // set mock to return ex  
   val result = ...
   ex shouldBe expectedError
}
```

Reply:
Fixed in https://github.com/hmrc/slack-notifications/pull/1/commits/e20830fe9ef3ee40faec5212d2ba968a323d8935